### PR TITLE
Add Verhoeff algorithm check/generate in chip-tool

### PR DESCRIPTION
**Pull request by krypton36**
#### Problem
Add Verhoeff algorithm check/generate for the following issue #13025

#### Change overview
- Adds check/generate of the Verhoeff from lib/support

#### Testing
Tested using chip-tool and generated manual pairing code.

### CODE REVIEW

File: examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+CHIP_ERROR SetupPayloadVerhoeffVerify::Run()
+{
+    std::string codeString(mSetupCode);
+    bool result;
+
+    bool isQRCode = SetupPayloadParseCommand::IsQRCode(codeString);
+
+    VerifyOrReturnError(!isQRCode, CHIP_ERROR_NOT_IMPLEMENTED);
+    VerifyOrReturnError(codeString.length() > mPos && mPos >= 0, CHIP_ERROR_INVALID_STRING_LENGTH);
```
Comment (andy31415) [inline examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp:34]: could we do the codeString length check before calling IsQRCode? Ordering wise it looks like this is the simplest and should be the first thing to validate.

File: examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+{
+    char checkChar;
+
+    checkChar = codeString.at(mPos);
+    codeString.erase(mPos, mPos);
+    ChipLogDetail(SetupPayload, "Verifying Manual Code: %s", codeString.c_str());
+
+    result = Verhoeff10::ValidateCheckChar(checkChar, codeString.c_str());
+
+    return CHIP_NO_ERROR;
```
Comment (andy31415) [inline examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp:52]: this function can never return anything else than CHIP_NO_ERROR. How about making it return the bool result instead?

File: examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    ReturnErrorOnFailure(GenerateChar(codeString, generatedChar));
+    ChipLogProgress(SetupPayload, "Generated Char: %c", generatedChar);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SetupPayloadVerhoeffGenerate::GenerateChar(std::string codeString, char & generatedChar)
+{
+    bool isQRCode = SetupPayloadParseCommand::IsQRCode(codeString);
+    ;
```
Comment (andy31415) [inline examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp:69]: remove extra `;`

File: examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    VerifyOrReturnError(!isQRCode, CHIP_ERROR_NOT_IMPLEMENTED);
```
Comment (andy31415) [inline examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp:71]: should this not be some form of 'INVALID_ARGUMENT' ?

File: examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp (hunk 1 above)
Comment (krypton36) [reply examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp:34]: Thank you for the review. Updated.

File: examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp (hunk 2 above)
Comment (krypton36) [reply examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp:52]: Updated.

File: examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp (hunk 3 above)
Comment (krypton36) [reply examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp:69]: Fixed.

File: examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp (hunk 4 above)
Comment (krypton36) [reply examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp:71]: Updated.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**